File: solutions/java/src/coffeevendingmachine/CoffeeRecipe.java
package coffeevendingmachine;

import java.util.Map;

public class CoffeeRecipe {
    private final String name;
    private final double price;
    private final Map<String, Integer> recipe;

    public CoffeeRecipe(String name, double price, Map<String, Integer> recipe) {
        this.name = name;
        this.price = price;
        this.recipe = recipe;
    }

    public String getName() {
        return name;
    }

    public double getPrice() {
        return price;
    }

    public Map<String, Integer> getRecipe() {
        return recipe;
    }
}

File: solutions/typescript/src/CoffeeVendingMachine/CoffeeRecipe.ts
class CoffeeRecipe {
  private name: string;
  private price: number;
  private recipe: Map<string, number>;

  constructor(name: string, price: number, recipe: Map<string, number>) {
    this.name = name;
    this.price = price;
    this.recipe = recipe;
  }

  public getName(): string {
    return this.name;
  }

  public getPrice(): number {
    return this.price;
  }

  public getRecipe() {
    return this.recipe;
  }
}
export default CoffeeRecipe;
